lib/Dialect/TTNN/Transforms/Passes.cpp

```diff
@@ -98,6 +111,66 @@ class TTNNDeallocate : public impl::TTNNDeallocateBase<TTNNDeallocate> {
   }
 };
 
+class TTNNValidateGraphCapture
+    : public impl::TTNNValidateGraphCaptureBase<TTNNValidateGraphCapture> {
+
+public:
+  using impl::TTNNValidateGraphCaptureBase<
+      TTNNValidateGraphCapture>::TTNNValidateGraphCaptureBase;
+
+  void runOnOperation() final {
+    const std::filesystem::path tmpDirPath =
+        std::filesystem::temp_directory_path();
+
+    const std::string mlirFilePath = tmpDirPath / "module.mlir";
+    const std::string flatBufferFilePath = tmpDirPath / "module.ttnn";
+    const std::string outReportPath = tmpDirPath / "module_graph_capture.json";
+
+    outputTTNNIRFile(mlirFilePath);
+    outputFlatBufferFile(mlirFilePath, flatBufferFilePath);
+    runGraphCapture(flatBufferFilePath, outReportPath);
+
+    if (!isValidGraphCaptureReport(outReportPath)) {
+      // TODO (nobradovic/odjuricic): Handle recompile.
+    }
+  }
+
+  void outputTTNNIRFile(const std::string &mlirFilePath) {
+    ModuleOp module = getOperation();
+    std::error_code _ec;
+    auto fs = llvm::raw_fd_stream(mlirFilePath, _ec);
+    module.print(fs);
+  }
+
+  void outputFlatBufferFile(const std::string &mlirFilePath,
+                            const std::string &flatBufferFilePath) {
+    const std::string cmd =
+        "./build/bin/ttmlir-translate --ttnn-to-flatbuffer " + mlirFilePath +
+        " -o " + flatBufferFilePath;
+
+    system(cmd.c_str());
+  }
+
+  void runGraphCapture(const std::string &flatBufferFilePath,
+                       const std::string &outReportFilePath) {
+    // TODO(mbezulj): Add required env variable to be able to run graph capture
+    // with mockup device and without kernel compilation.
+    const std::string cmd = "ttrt run " + flatBufferFilePath +
```

Comment:
Do we have something like std::format to allow plugin of parameters within const string?